pkg/engine/logging/logger.go

```diff
@@ -0,0 +1,25 @@
+package logging
+
+var Singleton Logger
+
+type Logger interface {
+	Log(e interface{})
+	Flush() string
```

Comment:
thinking out loud, should `Flush` even be something in this interface? If I am a test or the cli or wasm, I should know what logger instance I have and what other methods it has, if any. For a solution where we are flushing during the `Log` call, we wouldn't need a `Flush() string` method